Source/Python/WarpX.cpp

```diff
@@ -90,6 +126,26 @@ void init_WarpX (py::module& m)
             "Evolve the simulation the specified number of steps"
         )
 
+        .def_property("omp_threads",
+            [](WarpX & /* wx */){
+                return detail::get_or_throw<std::string>("amrex", "omp_threads");
+            },
+            [](WarpX & /* wx */, std::variant<int, std::string> omp_threads_var) {
+                std::visit([&]( auto && omp_threads) {
+                    amrex::ParmParse pp_amrex("amrex");
+                    pp_amrex.add("omp_threads", omp_threads);
+
+                    // set the value if not "system" or "nosmt"
+                    if constexpr(std::is_same_v<std::decay_t<decltype(omp_threads)>, int>) {
+                        amrex::Print() << "Changing WarpX threads to N=" << omp_threads << "\n";
+                        amrex::OpenMP::set_num_threads(omp_threads);
+                    }
+                }, omp_threads_var);
```

Comment:
@lucafedeli88 in some cases, the setter is a bit more complex, e.g., calling an extra function to make sure an option goes into effect immediately. Should also go into the get/set param parse logic files we want to introduce.

Reply:
Yes, I agree! In this case the setter is very simple, but in other cases we will need to call extra functions to ensure that an option has the intended effect (e.g., switching an algorithm may require to change the number of guard cells). I am not sure about what is the best architecture to achieve this cleanly (e.g., does the setter call the appropriate function or do certain containers in WarpX "observe" a set of parameters in order to act as soon as they are changed).  However, as you said, in this case it is very simple!